File: agent-driven-architecture-main/examples/typescript/src/spine/pure/ids.ts
// ── spine/pure/ids — value types (G4: the purity boundary is a folder) ──────
// ZERO I/O, zero logic. These are the names the whole system agrees on.
//
// NOTE (6.8): `ToolName` is a bare alias, NOT a branded type. The tool name is
// the discriminant of both `ToolResult` and `Command`, and a branded type
// cannot be a discriminant — a literal `"setPriority"` would not be assignable
// to it. One name per verb, and that name is a plain string.

export type ToolName = string;
export type CommandId = string;
export type Timestamp = number;
export type StepIndex = number;
export type SessionId = string;

// The raw, undecoded payload that arrives from a surface or a model. It is the
// one OPEN input in the system (6.10); the boundary closes it via a Verb's
// declared schema before anything downstream sees it.
export type RawInput = unknown;

// Shared identifier vocabulary. These are strings, not entities; several blocks
// name the same ticket without any of them naming each other (G11).
export type TicketId = string;
export type PanelId = string;

// Where an off-bus input came from (12.2). It is the key conflation counts by,
// the scope dedupe runs in, and the attribution an Interrupt or a Drain carries
// — so it is a first-class name, not a label someone remembers to set.
export type SourceName = string;

// The identity of ONE item on a durable queue. A durable source dedupes on it;
// a perishable source ignores it. Never null, so there is no branch.
export type SourceKey = string;

File: agent-driven-architecture-main/examples/kotlin/spine/src/main/kotlin/adr/spine/pure/Ids.kt
// ── spine/pure/ids — the value vocabulary every seam speaks ─────────────────
// ZERO I/O, zero behaviour. These are the names the sealed transport hierarchies
// are built out of, so a String can never be mistaken for a tool name and an
// index can never be mistaken for a timestamp.
//
// Why the ticket/panel ids live in the SPINE and not in a block: three blocks
// (triage, escalation, console) each hold a slice keyed by TicketId, and G11
// forbids a block naming a sibling. A shared opaque id type is the only legal
// home for a vocabulary two blocks must both be able to spell. It carries no
// domain logic — that stays in the owning block.

package adr.spine.pure


/** ONE NAME PER VERB (6.8): the discriminant of ToolResult, of Command, and the registry key. */
@JvmInline
value class ToolName(val value: String)

/** Minted ONLY at the boundary, from the committed sequence (G9). */
@JvmInline
value class CommandId(val value: String)

/** Read ONLY from the injected Clock, exactly once per step (G9). */
@JvmInline
value class Timestamp(val value: Long)

/** The offset the bus returned when a step was committed — the origin of every EffectKey (G9). */
@JvmInline
value class StepIndex(val value: Int)

/** The unit of work the stream is scoped to (5.2) — the only bound a folded budget may claim (G6). */
@JvmInline
value class SessionId(val value: String)

/** An opaque support-ticket identifier. */
@JvmInline
value class TicketId(val value: String)

/** An opaque console-panel identifier. */
@JvmInline
value class PanelId(val value: String)

/**
 * WHERE an off-bus input came from: a ticket stream, a sensor feed, a peer tier's
 * relay, an operator's console. It is the key conflation counts by, the scope
 * dedupe runs in, and the attribution an Interrupt or a Drain carries (12.2).
 */
@JvmInline
value class SourceName(val value: String)

/**
 * The source's OWN identifier for one work item — a queue message id, a webhook
 * delivery id. A durable policy dedupes on it; a perishable policy ignores it.
 * Never minted here: it belongs to the source, which is why it is opaque.
 */
@JvmInline
value class SourceKey(val value: String)

/** A relative duration. Never a clock reading — deadlines are handed in, not read (G9). */
typealias Millis = Long
